src/client/datascience/variablesView/notebookWatcher.ts

```diff
@@ -62,14 +69,26 @@ export class NotebookWatcher implements INotebookWatcher {
         this.notebookExtensibility.onKernelStateChange(this.kernelStateChanged, this, this.disposables);
         this.notebookEditorProvider.onDidChangeActiveNotebookEditor(this.activeEditorChanged, this, this.disposables);
         this.notebookEditorProvider.onDidCloseNotebookEditor(this.notebookEditorClosed, this, this.disposables);
+        notebook.onDidChangeCellExecutionState(this.onDidChangeCellExecutionState, this, this.disposables);
+    }
+
+    // Handle when a cell finishes execution
+    private onDidChangeCellExecutionState(cellStateChange: NotebookCellExecutionStateChangeEvent): void {
+        // If a cell has moved to idle, update our state
+        if (cellStateChange.executionState === NotebookCellExecutionState.Idle) {
+            // Convert to the old KernelStateEventArgs format
+            this.handleExecute({
```

Comment:
Cleanest code change was to just convert to old style event here. It's never silent as vscode API cell executions only cover user cell executions, not our silent cell init executions.